packages/environment-check/src/checks/destination.ts

```diff
@@ -0,0 +1,248 @@
+): Promise<{ messages: ResultMessage[]; destinationResults: DestinationResults }> {
+    const logger = getLogger();
+    logger.log(t('checkingDestination', { destination: destination.name }));
+
+    const { messages, result: destinationResults } = await checkCatalogServices(destination, username, password);
+
+    logger.push(...messages);
+    return {
+        messages: logger.getMessages(),
+        destinationResults
+    };
+}
+
+/**
+ * Checks for services from catalog requests.
+ *
+ * @param destination sap detination
+ * @param username username
+ * @param password password
+ * @returns Result messages and results of catalog requests
+ */
+async function checkCatalogServices(
+    destination: Destination,
+    username?: string | undefined,
+    password?: string | undefined
+): Promise<{ messages: ResultMessage[]; result: DestinationResults }> {
+    const messages: ResultMessage[] = [];
+
+    const v2results = await catalogRequest(ODataVersion.v2, destination, username, password);
+    messages.push(...v2results.messages);
+
+    const v4results = await catalogRequest(ODataVersion.v4, destination, username, password);
+    messages.push(...v4results.messages);
+
+    const html5DynamicDestination = !!destination.basProperties?.html5DynamicDestination;
+
+    if (!html5DynamicDestination) {
+        messages.push({
+            severity: Severity.Error,
+            text: t('error.missingDynamicDestProperty', { destination: destination.name })
+        });
+    }
+
+    const result: DestinationResults = {
+        v2: { results: v2results.result, status: v2results.responseStatus },
+        v4: { results: v4results.result, status: v4results.responseStatus },
+        HTML5DynamicDestination: html5DynamicDestination
+    };
+
+    return {
+        messages,
+        result
+    };
+}
+
+/**
+ * Performs a catalog request for the given odata version and destination.
+ *
+ * @param odataVersion odataVersion to be used
+ * @param destination destination to be checked
+ * @param username username
+ * @param password password
+ * @returns messages, catalog results, response status
+ */
+async function catalogRequest(
+    odataVersion: ODataVersion,
+    destination: Destination,
+    username?: string | undefined,
+    password?: string | undefined
+): Promise<{ messages: ResultMessage[]; result: CatalogResultV2 | CatalogResultV4; responseStatus: number }> {
+    const logger = getLogger();
+    let result: CatalogResultV2 | CatalogResultV4;
+    let url: string;
+    let responseStatus: number;
+    try {
+        const auth =
+            username !== undefined && password !== undefined
+                ? {
+                      username,
+                      password
+                  }
+                : undefined;
+
+        const axiosConfig: AxiosRequestConfig = {
+            baseURL: destination.host,
+            auth: auth
+        };
+
+        const provider: AbapServiceProvider = createForAbap(axiosConfig);
```

Comment:
You should use the `createForDestination` function when working with destinations in BAS. 

To test:

- Go to folder examples/odata-cli
- Create a folder .env containing a file called env.properties
- Add TEST_DESTINATION=YOUR_DESTINATION to the env.properties file
- Execute pnpm start btp
- Check the .tmp folder for the results

 
The relevant code for you is at https://github.com/SAP/open-ux-tools/blob/main/examples/odata-cli/src/index.ts#L195-L206

Reply:
Now using createForDestination